Use C++20 std::ranges::sort in WTF & Tests/WTF

iidmsa opened the pull request:
#### 5cf50dca5844d48388cdfe335e257c09944e0bca
<pre>
Use C++20 std::ranges::sort in WTF &amp; Tests/WTF
<a href="https://bugs.webkit.org/show_bug.cgi?id=302755">https://bugs.webkit.org/show_bug.cgi?id=302755</a>
<a href="https://rdar.apple.com/165012566">rdar://165012566</a>

Reviewed by Darin Adler and Sam Weinig.

Test: Tools/TestWebKitAPI/Tests/WTF/IntervalSet.cpp
* Source/WTF/wtf/Liveness.h:
(WTF::Liveness::LiveAtHead::LiveAtHead):
* Source/WTF/wtf/RangeSet.h:
* Tools/TestWebKitAPI/Tests/WTF/IntervalSet.cpp:
(TestWebKitAPI::stressTest):

Canonical link: <a href="https://commits.webkit.org/303372@main">https://commits.webkit.org/303372@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/08f52edc29458bda18150edcbaa084e978a89eee

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131968 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/4460 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/42982 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139481 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/83865 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/133838 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/4402 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/4222 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/100872 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/68265 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134914 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/3135 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/118186 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81663 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/3029 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/887 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82700 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/124031 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/111781 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/36310 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/142126 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/130475 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/4130 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/36890 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/109244 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/4211 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/3737 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109414 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27762 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/3123 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114466 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/57351 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/4183 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/32863 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/163442 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/4015 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/67630 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/42491 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/4275 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/4143 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

weinig commented on Source/WTF/wtf/RangeSet.h:
```diff
@@ -155,11 +155,9 @@ class RangeSet final {
-            [&] (const Range& a, const Range& b) -> bool {
-                return a.begin() < b.begin();
-            });
+        std::ranges::sort(m_ranges, [&] (const Range& a, const Range& b) -> bool {
```
Please remove the space between the `[&]` and the parameter list. You can also remove the return type here.

```suggestion
        std::ranges::sort(m_ranges, [&](const Range& a, const Range& b) {
```

iidmsa replied:
Done.

iidmsa committed:
Use C++20 std::ranges::sort in WTF & Tests/WTF
https://bugs.webkit.org/show_bug.cgi?id=302755
rdar://165012566

Reviewed by Darin Adler and Sam Weinig.

Test: Tools/TestWebKitAPI/Tests/WTF/IntervalSet.cpp
* Source/WTF/wtf/Liveness.h:
(WTF::Liveness::LiveAtHead::LiveAtHead):
* Source/WTF/wtf/RangeSet.h:
* Tools/TestWebKitAPI/Tests/WTF/IntervalSet.cpp:
(TestWebKitAPI::stressTest):

Canonical link: https://commits.webkit.org/303372@main
